test/e2e/e2e_test.go

```diff
@@ -169,16 +171,58 @@ var _ = ginkgo.ReportAfterSuite("Kubernetes e2e suite report", func(report ginkg
 			klog.Errorf("Error in detail reporter: %v", err)
 			return
 		}
-		_, err = f.Write(b)
-		if err != nil {
+		if _, err = f.Write(b); err != nil {
 			klog.Errorf("Error saving test details in detail reporter: %v", err)
 			return
 		}
 		// Printing newline between records for easier viewing in various tools.
-		_, err = fmt.Fprintln(f, "")
-		if err != nil {
+		if _, err = fmt.Fprintln(f, ""); err != nil {
 			klog.Errorf("Error saving test details in detail reporter: %v", err)
 			return
 		}
 	}
+}
+
+// writeJUnitReport generates a JUnit file in the e2e report directory that is
+// shorter than the one normally written by `ginkgo --junit-report`. This is
+// needed because the full report can become too large for tools like Spyglass
+// (https://github.com/kubernetes/kubernetes/issues/111510).
+//
+// Users who want the full report can use `--junit-report`.
+func writeJUnitReport(report ginkgo.Report) {
+	if framework.TestContext.ReportDir == "" {
+		return
+	}
+
+	trimmedReport := report
+	trimmedReport.SpecReports = nil
+	for _, specReport := range report.SpecReports {
+		// Remove details for any spec that hasn't failed. In Prow,
+		// the test output captured in build-log.txt has all of this
+		// information, so we don't need it in the XML.
+		if specReport.State != types.SpecStateFailed {
+			// strip the "system-error" if the testcase is not failed.
+			specReport.CapturedGinkgoWriterOutput = ""
+			// strip the "system-out" if the testcase is not failed.
+			specReport.CapturedStdOutErr = ""
+			// strip some details for tracing each steps executed by Ginkgo, this is used to build the "system-out"
+			// while "system-out" is not shown by Spyglass if the testcase is not failed.
+			specReport.ReportEntries = nil
```

Comment:
Note that trimming this here is slightly too aggressive: if a test stores measurements during a successful run, then we probably should keep those entries because otherwise doing those measurements is questionable (where are they visible?).

I'm not sure whether this is a real problem and I will send a PR with my original code for trimming ReportEntries anyway as soon as this gets merged, so I don't mind having this go in as-is.

Reply:
> if a test stores measurements during a successful run, then we probably should keep those entries because otherwise doing those measurements is questionable

I wrote a simple test with measurements, the testcase passed without any issue, so I guess that entry added will only be used to generate a report, and while the testcase passed, those info is not shown by any downstream tools, so I think it should be fine.

+1 on more cleanup around this in a separate pr.

Reply:
Where could you see the result of the measurements?

Reply:
no, I can just see that testcase passed.

```
          <testcase name="[It] [sig-autoscaling] [Feature:ClusterSizeAutoscalingScaleUp] [Slow] Autoscaling Simple Check do nothing here should takes less than 1 minutes" classname="Kubernetes e2e suite" status="passed" time="0.063344167"></testcase>
```